src/lib/features/project-status/projects-status.e2e.test.ts

```diff
@@ -264,36 +264,46 @@ test('project status includes stale flags', async () => {
         {} as IUser,
         {} as IAuditUser,
     );
-    const createFlagInState = async (
-        name: string,
-        state?: Object,
-        projectId?: string,
-    ) => {
-        await app.createFeature(name, projectId);
-        if (state) {
-            await db.rawDatabase('features').update(state).where({ name });
-        }
-    };
 
-    await createFlagInState('stale-flag', { stale: true });
-    await createFlagInState('potentially-stale-flag', {
-        potentially_stale: true,
-    });
-    await createFlagInState('potentially-stale-and-stale-flag', {
-        potentially_stale: true,
-        stale: true,
-    });
-    await createFlagInState('non-stale-flag');
-    await createFlagInState('archived-stale-flag', {
-        archived: true,
-        stale: true,
-    });
-    await createFlagInState(
-        'stale-other-project',
-        { stale: true },
-        otherProject.id,
+    function cartesianProduct(...arrays: any[][]): any[][] {
+        return arrays.reduce(
+            (acc, array) => {
+                return acc.flatMap((accItem) =>
+                    array.map((item) => [...accItem, item]),
+                );
+            },
+            [[]] as any[][],
+        );
+    }
```

Comment:
We could also do 4 nested for loops if we prefer, but this solution feels descriptive enough to me, at least.